sources/packages/web/src/router/InstitutionRoutes.ts

```diff
@@ -1,25 +1,68 @@
 import { RouteRecordRaw } from "vue-router";
 import InstitutionDashboard from "../views/institution/InstitutionDashboard.vue";
 import AppInstitution from "../views/institution/AppInstitution.vue";
-import { InstitutionRoutesConst } from "../constants/routes/RouteConstants";
+import {
+  InstitutionRoutesConst,
+  SharedRouteConst,
+} from "../constants/routes/RouteConstants";
 import Login from "../views/institution/Login.vue";
+import WorkInProgress from "../views/WorkInProgress.vue";
+import { AppConfigService } from "../services/AppConfigService";
+import { ClientIdType } from "../types/contracts/ConfigContract";
+import { AuthStatus, AppRoutes } from "../types";
 
 export const institutionRoutes: Array<RouteRecordRaw> = [
   {
-    path: "/institution",
+    path: AppRoutes.InstitutionRoot,
     name: InstitutionRoutesConst.APP_INSTITUTION,
     component: AppInstitution,
     children: [
       {
-        path: "login",
+        path: AppRoutes.Login,
         name: InstitutionRoutesConst.LOGIN,
         component: Login,
       },
       {
-        path: "institution-dashboard",
+        path: AppRoutes.InstitutionDashboard,
         name: InstitutionRoutesConst.INSTITUTION_DASHBOARD,
         component: InstitutionDashboard,
       },
+      {
+        path: AppRoutes.InstitutionProfile,
+        name: InstitutionRoutesConst.INSTITUTION_PROFILE,
+        component: WorkInProgress,
+      },
     ],
+    beforeEnter: (to, from, next) => {
+      AppConfigService.shared
+        .initAuthService(ClientIdType.INSTITUTION)
+        .then(() => {
+          const status = AppConfigService.shared.authStatus({
+            type: ClientIdType.INSTITUTION,
+            path: to.path,
+          });
+          switch (status) {
+            case AuthStatus.Continue:
+              next();
+              break;
+            case AuthStatus.RequiredLogin:
+              next({
+                name: InstitutionRoutesConst.LOGIN,
+              });
+              break;
+            case AuthStatus.RedirectHome:
+              next({
+                name: InstitutionRoutesConst.INSTITUTION_DASHBOARD,
+              });
+              break;
+            case AuthStatus.ForbiddenUser:
```

Comment:
Maybe would be good to have a default option in this switch.
Would it be good to maybe consider changing the default one to be the SharedRouteConst.FORBIDDEN_USER.